src/crypto/iota_crypto.h

```diff
@@ -8,6 +8,38 @@
 #include <stdint.h>
 #include <stdlib.h>
 
+#ifdef CRYPTO_USE_SODIUM
```

Comment:
the intention of iota_crypto is to provide an interface between 3rd party crypto libs and iota client lib. in this way, the client lib can integrate with different crypto libs as needed. exporting these crypto headers breaks this advantage which means 3rd party crypto libs compatibility needs to be considered in Core/Client modules, like [here](https://github.com/iotaledger/iota.c/pull/363/files#diff-4f022eaaaadd51600447cc4544bb7942cac5aae3fd479477cfa4e9daecf8da4cR74-R78)

```
if defined(CRYPTO_USE_SODIUM)
  crypto_generichash_blake2b_state blake_state;
#elif defined(CRYPTO_USE_BLAKE2B_REF)
  blake2b_state blake_state;
#endif
```

Reply:
Thanks for the explanation. Crypto headers will be moved back to their previous place and the code will be refactored.